File: enode/src/main/java/org/enodeframework/queue/ISendReplyService.java
package org.enodeframework.queue;

import org.enodeframework.commanding.CommandResult;
import org.enodeframework.common.io.ReplySocketAddress;
import org.enodeframework.queue.domainevent.DomainEventHandledMessage;

import java.util.concurrent.CompletableFuture;

public interface ISendReplyService {
    CompletableFuture<Boolean> sendCommandReply(CommandResult replyData, ReplySocketAddress replyAddress);

    CompletableFuture<Boolean> sendEventReply(DomainEventHandledMessage replyData, ReplySocketAddress replyAddress);
}

File: enode/src/main/java/org/enodeframework/queue/ISendReplyService.kt
package org.enodeframework.queue

import org.enodeframework.commanding.CommandResult
import org.enodeframework.queue.domainevent.DomainEventHandledMessage
import java.util.concurrent.CompletableFuture

interface ISendReplyService {
    /**
     * Send command handle result
     */
    fun sendCommandReply(commandResult: CommandResult, address: String): CompletableFuture<Boolean>

    /**
     * Send event handle result
     */
    fun sendEventReply(eventHandledMessage: DomainEventHandledMessage, address: String): CompletableFuture<Boolean>
}